app/models/supervised_experiment.rb

```diff
@@ -1,63 +1,122 @@
+##
+# This class represents an instance of supervised experiment.
+# SupervisedExperiment is a subclass of CustomPointsExperiment, with is subclass
+# of Experiment. The main difference from normal experiment is that in case of
+# supervised experiment scheduling of new simulation is external action performed
+# by supervisor script maintaining by experiment supervisor. This type of experiment
+# have also result, which is result of script that supervisors this experiment e.g.
+# some value or point from input space.
+#
+# List of possible attributes:
+# * supervised - boolean, always true, marker which allow to distinguish normal experiment from
+#   supervised one [set by #initialize]
+# * completed - boolean, marker which allow to check if experiment is completed
+#   (which equals that supervisor script finished and sent result) [set false by #initialize,
+#   modified to true by #mark_as_complete!]
+# * results - hash, contains result of experiment (sent by supervisor script) [set empty by
+#   initialize, modified by #mark_as_complete!]
+# * supervisor_script_uuid - string, id of supervisor script, used for authentication [set by
+#   #start_supervisor_script]
+# * is_error - boolean, this flag is true when experiment is in error state
+#   [set by ExperimentController#mark_as_complete]
+# * error_reason - string, set with is_error flag is_reason [set by ExperimentController#mark_as_complete]
 class SupervisedExperiment < CustomPointsExperiment
 
+  ##
+  # Sets parameters needed for SupervisorExperiment (described above), and calls super.
+  # Must be call once on creation to proper initialization
   def init_empty(simulation)
     super simulation
     self.supervised = true
     self.completed = false
-    self.result = {}
+    self.results = {}
   end
 
-  def start_supervisor_script(user, supervisor_script_id, supervisor_script_params, experiment_input)
-    supervisor_script_params['experiment_id'] = self.id.to_s
+  ##
+  # Query only SupervisedExperiments from DB with additional conditions.
+  def self.where(cond, opts = {})
+    super({supervised: true}.merge(cond), opts)
+  end
```

Comment:
Co do init_empty, rzeczywiście przegapiłem że tam było odwołanie do super które szło do init_empty CustomPointExperiment. Zmieniłem to z powrotem na to co było wcześniej.

Co do where - dodałem tak jak sugerowałeś, przy czym moja uwaga że jeśli chcemy w pełni poprawnego musielibyśmy przeładować tak dużo więcej metod. Przykład z count (wiem że nie w pełni zainicjalizowane, ale w tym przypadku to nie ma znaczenia):

2.1.2 :003 > e = Experiment.new({})
 => #<Experiment:0x000000070508f0 @attributes={}> 
2.1.2 :004 > e.save
 => BSON::ObjectId('55277eea2ab4f2acbb000001') 
2.1.2 :005 > e = SupervisedExperiment.new({supervised: true})
 => #<SupervisedExperiment:0x00000006efd598 @attributes={"supervised"=>true}> 
2.1.2 :007 > e.save
 => BSON::ObjectId('55277f0b2ab4f2acbb000002') 
2.1.2 :008 > Experiment.count
 => 2 
2.1.2 :009 > SupervisedExperiment.count
 => 2